### PR TITLE
added sdbm-hash algorithm

### PR DESCRIPTION
**Pull request by christianbender**
I added the sdbm-hash algorithm as a function called **sdbm**

### CODE REVIEW

File: snippets/sdbm-hash.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+This function implements the **sdbm** hash-algorithm. 
+It uses the charCodeAt-methode and the shift-operator.
```
Comment (Chalarangelo) [inline snippets/sdbm-hash.md]: Please update the description to explain the methods and techniques used.

File: snippets/sdbm-hash.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+This function implements the **sdbm** hash-algorithm.
```
Comment (Chalarangelo) [inline snippets/sdbm-hash.md]: Please update the description to explain what the algorithm does.

File: snippets/sdbm-hash.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+This function implements the **sdbm** hash-algorithm. 
+It uses the charCodeAt-methode and the shift-operator.
+
+``` js
+function sdbm(s) {
```
Comment (Chalarangelo) [inline snippets/sdbm-hash.md]: Please follow the styleguide, as seen in CONTRIBUTING.md and snippet_template.

File: snippets/sdbm-hash.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    var hashCode = 0;
```
Comment (Chalarangelo) [inline snippets/sdbm-hash.md]: Please use `let` or `const` instead of `var`.

File: snippets/sdbm-hash.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    for (var i = 0; i < s.length; i++) {
+        hashCode = s.charCodeAt(i) + (hashCode << 6) + (hashCode << 16) - hashCode;
+    }
```
Comment (Chalarangelo) [inline snippets/sdbm-hash.md]: Prefer using `Array.map` or `Array.reduce` over loops, if possible.

File: snippets/sdbm-hash.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    return hashCode;
+}
+```
```
Comment (Chalarangelo) [inline snippets/sdbm-hash.md]: Please add examples with sample input and expected output in your snippet.

**Comment by christianbender** [comment]
@Chalarangelo  

hello, 

thanks for feedback. 
I will fix that style issues as fast as I can
It is a simple hash algorithm. see here.
http://www.cse.yorku.ca/~oz/hash.html
https://softwareengineering.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed

greetings

**Comment by christianbender** [comment]
@iamsoorena  

I don't know. I need to access to the last item I puted in the array. I considered over  the map-function. Has someone got an idea?

**Comment by kriadmin** [comment]
@christianbender This is how you can do it using `Array.reduce()`
[Array.reduce() MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) 
```js
const sdbm = str => {
     let arr = str.split('');
    return arr.reduce( (hashCode,currentVal) =>{
                 return  hashCode = currentVal.charCodeAt(0) + (hashCode << 6) + (hashCode << 16)  - hashCode;
     }
 ,0)
 }

**Comment by christianbender** [comment]
@kriadmin Thanks! Why didn't I come up with this myself?

**Pull request merged**